### ISSUE TITLE
Toolbar Searchfield in CAP is not rendered

### ISSUE THREAD

**Issue by anhallbe** [infor-design/enterprise-ng#594]
**Describe the bug**
When a CAP header contains a toolbar with a searchfield in it, the searchfield is not rendered.

When modifying the code by removing the `class="toolbar"`, the searchfield is rendered but an extra toolbar appears.

**To Reproduce**
1. Check out the "searchfield-5.5.2" branch of https://github.com/anhallbe/enterprise-ng
2. Run `npm install` if needed, and then `npm start`
3. Open the "Contextual Action Panel" sample
4. Click the "Panel with Searchfield" button
5. ISSUE A: The panel is opened, but no searchfield is rendered.
6. Close the CAP
7. Click the "Panel with Searchfield in flex toolbar button"
8. ISSUE B: The panel is opened, but it has an extra toolbar with the default CAP title.

**Expected behavior**
At 5) there should be a searchfield icon that can be expanded to a searchfield.
At 8) there should only be one toolbar.

**Version**
- ids-enterprise-ng: 5.5.2
- ids-enterprise: 4.20.1

**Screenshots**
![cap-searchfield](https://user-images.githubusercontent.com/4991954/63922153-6bff6f80-ca44-11e9-947a-f0c3044eb903.gif)

![image](https://user-images.githubusercontent.com/4991954/63921750-afa5a980-ca43-11e9-9a88-43e3c7503006.png)

**Platform**
 - OS Version: Windows 10
 - Browser Name: Chrome, Firefox

**Additional context**
ISSUE A as described above can be resolved by not explicitly setting the `toolbar` class on `soho-toolbar`. However when doing so, the issue described as ISSUE B occurs, regardless of whether `soho-toolbar` or `soho-toolbar-flex` is used.
Code: https://github.com/anhallbe/enterprise-ng/tree/searchfield-5.5.2/src/app/contextual-action-panel

**Comment by EdwardCoyle** [comment]
Paging @Sovia, who may also be able to provide some input on changes on the app side.  

We had a similar issue crop up in the Landmark platform after the memory leak changes made in infor-design/enterprise#2496.  We did patch this in infor-design/enterprise#2691, but we didn't make any changes in the context of a CAP.  As far as I know, these patches fixed the Landmark issue.

One thing to note is that before we fixed the memory leaks, the lifecycle for Toolbars with `noSearchfieldReinvoke` set to true wasn't completely right.  With the improvements (which fix a pretty bad DOM leak in SPAs), if `noSearchfieldReinvoke` is true, it will be necessary on the app side to make sure a Soho Toolbar Searchfield is manually setup and torn down.
